robyn/argument_parser.py

```diff
@@ -23,7 +23,7 @@ def __init__(self) -> None:
             "--dev",
             dest="dev",
             action="store_true",
-            default=False,
+            default=None,
```

Comment:
Let's document this decision?

Reply:
@sansyrox It will be defaulted to false if an environment variable is not found. (since the fallback value to `os.getenv` on `cli#run` is false) Should we still document it?

Reply:
@VishnuSanal , I want you to document the decision that the env variable will have a priority over the cli flag. 

I believe we should prioritise cli flag over the env variable.